File: _more/compiler/py2rs/_version/v1/test/factorial_while.c
#include <stdio.h>
#include <stdlib.h>

int factorial_while(int n) {
    int result;
    result = 1;
    while (n > 1) {
    result = (result * n);
    n = (n - 1);
    }
    return result;
}

int main() {
    int result = factorial_while(5);
    printf("factorial_while(5)=%d\n", result);
    return 0;
}

File: _more/compiler/py2rs/_version/v1/test/factorial_while.rs
use std::collections::HashMap;
fn factorial_while(mut n: i32) -> i32 {
    let mut result = 1;
    while n > 1 {
        result = (result * n);
        n = (n - 1);
    }
    return result;
}

fn main() {
    let result = factorial_while(5);
    println!("factorial_while(5)={}", result);
}